internal/reconcilers/pipeline_git_repo.go

```diff
@@ -52,31 +55,65 @@ func (r *PipelineGitRepoReconciler) Reconcile(_ context.Context, pipeline *v1alp
 	return false, nil
 }
 
-func cloneOrPullRepo(targetPath string, branchName string) (string, error) {
+func cloneOrPullRepo(targetPath string, pipelineRelease string) (string, error) {
 	// Try to clone first
-	_, err := git.PlainClone(targetPath, false, &git.CloneOptions{
-		URL:           operatorPipelinesRepo,
-		ReferenceName: plumbing.NewBranchReferenceName(branchName),
+	r, err := git.PlainClone(targetPath, false, &git.CloneOptions{
+		URL: operatorPipelinesRepo,
 	})
 	if err != nil && err != git.ErrRepositoryAlreadyExists {
 		return "", err
 	}
 	// The directory is already there, so let's just update to latest
-	r, err := git.PlainOpen(targetPath)
-	if err != nil {
+	if r == nil && err == git.ErrRepositoryAlreadyExists {
+		var err error
+		r, err = git.PlainOpen(targetPath)
+		if err != nil {
+			return "", err
+		}
+	}
+
+	// If r is still both clone and open were unsuccessful, returning err to avoid a later panic
+	if r == nil {
+		return "", errors.New("could not clone or open repo")
```

Comment:
Rather than import errors, and have to change all the other errors, use `fmt.Errorf`